### PR TITLE
[ASD-2483] Validate folder name on change

### PR DESCRIPTION
**Pull request by eromano**
**Please check if the PR fulfills these requirements**

> -  [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the new behaviour?**

notify service refactoring 
get translate eliminitation in favor of instant
add error event where necessary
fix config problem during test

When an error in received on the creation of the folder an event will be raised 

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/directives/node-delete.directive.ts [hunk 1]
```diff
@@ -174,27 +168,23 @@ export class NodeDeleteDirective implements OnChanges {
         );
     }
 
-    private notify(status) {
-        this.getMessage(status).subscribe((message) => this.notification.openSnackMessage(message));
-    }
-
     private getMessage(status): Observable<string> {
```
Comment (DenisVuyka) [inline lib/core/directives/node-delete.directive.ts]: return type is not valid anymore I think, should be just `string`?

File: lib/core/directives/node-restore.directive.spec.ts [hunk 2]
```diff
@@ -76,7 +76,7 @@ describe('NodeRestoreDirective', () => {
     }));
 
     beforeEach(() => {
-        spyOn(translation, 'get').and.returnValue(Observable.of('message'));
+        spyOn(translation, 'instant').and.returnValue(Observable.of('message'));
```
Comment (DenisVuyka) [inline lib/core/directives/node-restore.directive.spec.ts]: "instant" returns a string value without observable

**Comment by DenisVuyka** [comment]
@eromano couple of comments on the return types

File: lib/core/directives/node-delete.directive.ts (hunk 1 above)
Comment (popovicsandras) [inline lib/core/directives/node-delete.directive.ts]: wrong return type

File: lib/core/directives/node-restore.directive.spec.ts (hunk 2 above)
Comment (popovicsandras) [inline lib/core/directives/node-restore.directive.spec.ts]: No Observable here

**Pull request merged**